oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -0,0 +1,289 @@
+import os
+import re
+import shutil
+import time
+from pathlib import Path
+
+import pandas as pd
+from Bio.Blast.Applications import NcbiblastnCommandline, NcbimakeblastdbCommandline
+from joblib import Parallel, delayed, parallel_backend
+
+from ._filter_base import ProbeFilterBase
+from ._utils import _load_probes_info, _write_output, _write_removed_genes
+
+
+class ProbeFilterBlastn(ProbeFilterBase):
+    def __init__(
+        self,
+        n_jobs,
+        file_transcriptome_fasta,
+        dir_output,
+        dir_annotations,
+        word_size,
+        percent_identity,
+        coverage,
+        probe_length_min,
+        probe_length_max,
+        ligation_region=0,
+    ):
+        """This class filters probes based on the blast alignment tool
+
+        :param file_transcriptome_fasta: path to fasta file containing all probes
+        :type file_transcriptome_fasta: str
+        :param word_size: word size for the blastn seed (exact match to target)
+        :type word_size: int
+        :param percent_identity: maximum similarity between probes and target sequences, ranging from 0 to 100% (no missmatch)
+        :type percent_identity: int
+        :param coverage: minimum coverage between probes and target sequence, ranging from 0 to 100% (full coverage)
+        :type coverage: int
+        :param probe_length_min: minimum length of the probes created
+        :type probe_length_min: int
+        :param probe_length_max: maximum length of the probes created
+        :type probe_length_max: int
+        :param ligation_region: coverage between probes and target sequence should not span region around ligation site (e.g. ligation_region = 5 would correspond to -4 to +5 nt around ligation site), if ligation_region = 0, omit this requirement
+        :type ligation_region: int
+        """
+        super().__init__(n_jobs, dir_output, dir_annotations)
+
+        self.word_size = word_size
+        self.percent_identity = percent_identity
+        self.coverage = coverage
+        self.file_transcriptome_fasta = file_transcriptome_fasta
+        self.probe_length_min = probe_length_min
+        self.probe_length_max = probe_length_max
+        self.ligation_region = ligation_region
+
+        self.dir_blast = os.path.join(self.dir_output, "blast")
+        Path(self.dir_blast).mkdir(parents=True, exist_ok=True)
+
+        self.dir_probes = os.path.join(self.dir_output, "probes_blast")
+        Path(self.dir_probes).mkdir(parents=True, exist_ok=True)
+
+        self.file_removed_genes = os.path.join(
```

Comment:
I think it would be more convenient if we don't write this into a file but store it in the database. We then can have a function that writes all genes without any probes from the database.